api/admin.go

```diff
@@ -390,9 +405,79 @@ func (a *API) adminUserDelete(w http.ResponseWriter, r *http.Request) error {
 			return internalServerError("Error recording audit log entry").WithInternalError(terr)
 		}
 
-		if terr := tx.Destroy(user); terr != nil {
-			return internalServerError("Database error deleting user").WithInternalError(terr)
+		if params.IsSoftDelete == "true" {
+			if user.DeletedAt != nil {
+				// user has been soft deleted already
+				return nil
+			}
+			softDeleteId, terr := crypto.GenerateNanoId(5)
+			if terr != nil {
+				return terr
+			}
+			user.Email = storage.NullString(fmt.Sprintf("%s-%x", softDeleteId, sha256.Sum256([]byte(user.GetEmail()))))
+			user.Phone = storage.NullString(fmt.Sprintf("%s-%x", softDeleteId, sha256.Sum256([]byte(user.GetPhone()))))
+			user.EmailChange = fmt.Sprintf("%s-%x", softDeleteId, sha256.Sum256([]byte(user.EmailChange)))
+			user.PhoneChange = fmt.Sprintf("%s-%x", softDeleteId, sha256.Sum256([]byte(user.PhoneChange)))
```

Comment:
Would you mind removing the password as well -- to null or `''` or whatever makes most sense.